src/ws/test-remotectlcertificate.c

```diff
@@ -130,11 +131,16 @@ setup (TestCase *tc,
   g_ptr_array_add (ptr, "--user");
   g_ptr_array_add (ptr, (gchar *) g_get_user_name ());
 
-  gr = getgrnam (g_get_user_name ());
-  if (gr != NULL)
+  /* determine user's primary group */
+  pwd = getpwnam (g_get_user_name ());
+  if (pwd)
     {
-      g_ptr_array_add (ptr, "--group");
-      g_ptr_array_add (ptr, (gchar *) g_get_user_name ());
+      struct group *gr = getgrgid (pwd->pw_gid);
```

Comment:
This seems pretty sideways, particularly since we need to scan `/etc/group` to get the `pw_gid` value in the first place...

Why not just go directly to `getgrgid (getgid ())` ?

Reply:
Right, this isn't anywhere near suid. Done, thanks for pointing out!